### <a name="change-in-behavior-for-taskwaitall-methods-with-time-out-arguments"></a><span data-ttu-id="48732-101">제한 시간 인수가 있는 Task.WaitAll 메서드에 대 한 동작 변경 사항</span><span class="sxs-lookup"><span data-stu-id="48732-101">Change in behavior for Task.WaitAll methods with time-out arguments</span></span>

|   |   |
|---|---|
|<span data-ttu-id="48732-102">설명</span><span class="sxs-lookup"><span data-stu-id="48732-102">Details</span></span>|<span data-ttu-id="48732-103">Task.WaitAll 수행 보다 일관 된.NET 4.5.In에.NET Framework 4에서는 이러한 메서드 되지 않게 동작 했었습니다.</span><span class="sxs-lookup"><span data-stu-id="48732-103">Task.WaitAll behavior was made more consistent in .NET 4.5.In the .NET Framework 4, these methods behaved inconsistently.</span></span> <span data-ttu-id="48732-104">제한 시간이 만료되고 메서드 호출 전에 하나 이상의 작업이 완료되거나 취소되면 이 메서드는 <xref:System.AggregateException?displayProperty=name> 예외를 throw합니다.</span><span class="sxs-lookup"><span data-stu-id="48732-104">When the time-out expired, if one or more tasks were completed or canceled before the method call, the method threw an <xref:System.AggregateException?displayProperty=name> exception.</span></span> <span data-ttu-id="48732-105">제한 시간이 만료되고 메서드 호출 전에 완료되거나 취소된 작업은 없지만 메서드 호출 후에 하나 이상의 작업이 완료되거나 취소되면 이 메서드는 false를 반환합니다.</span><span class="sxs-lookup"><span data-stu-id="48732-105">When the time-out expired, if no tasks were completed or canceled before the method call, but one or more tasks entered these states after the method call, the method returned false.</span></span><br/><br/><span data-ttu-id="48732-106">.NET Framework 4.5에서 이러한 메서드 오버 로드 이제 false를 반환 하는 경우 시간 제한 간격이 만료 되 고 throw 하는 경우 모든 작업이 실행 되는 <xref:System.AggregateException?displayProperty=name> (에 관계 없이 앞 이나 뒤의 메서드는 여부 입력된 한 작업이 취소 되는 경우에 예외 호출) 다른 작업이 실행 되 고 있습니다.</span><span class="sxs-lookup"><span data-stu-id="48732-106">In the .NET Framework 4.5, these method overloads now return false if any tasks are still running when the time-out interval expired, and they throw an <xref:System.AggregateException?displayProperty=name> exception only if an input task was cancelled (regardless of whether it was before or after the method call) and no other tasks are still running.</span></span>|
|<span data-ttu-id="48732-107">제안 해결 방법</span><span class="sxs-lookup"><span data-stu-id="48732-107">Suggestion</span></span>|<span data-ttu-id="48732-108">경우는 <xref:System.AggregateException?displayProperty=name> 코드 IsCanceled 속성을 통해 동일한 검색을 수행 해야 대신 호출 되는 WaitAll 호출 하기 전에 취소 된 작업을 검색 하는 방법으로 발견 되 고 되었습니다 (예:. Any(t =&gt; t.IsCanceled)) 이후 시간 제한 전에 대기 중이 던된 모든 작업을 완료 하는.NET 4.6이 경우 throw만 합니다.</span><span class="sxs-lookup"><span data-stu-id="48732-108">If an <xref:System.AggregateException?displayProperty=name> was being caught as a means of detecting a task that was cancelled prior to the WaitAll call being invoked, that code should instead do the same detection via the IsCanceled property (for example: .Any(t =&gt; t.IsCanceled)) since .NET 4.6 will only throw in that case if all awaited tasks are completed prior to the timeout.</span></span>|
|<span data-ttu-id="48732-109">범위</span><span class="sxs-lookup"><span data-stu-id="48732-109">Scope</span></span>|<span data-ttu-id="48732-110">부</span><span class="sxs-lookup"><span data-stu-id="48732-110">Minor</span></span>|
|<span data-ttu-id="48732-111">버전</span><span class="sxs-lookup"><span data-stu-id="48732-111">Version</span></span>|<span data-ttu-id="48732-112">4.5</span><span class="sxs-lookup"><span data-stu-id="48732-112">4.5</span></span>|
|<span data-ttu-id="48732-113">형식</span><span class="sxs-lookup"><span data-stu-id="48732-113">Type</span></span>|<span data-ttu-id="48732-114">런타임</span><span class="sxs-lookup"><span data-stu-id="48732-114">Runtime</span></span>|
|<span data-ttu-id="48732-115">영향을 받는 API</span><span class="sxs-lookup"><span data-stu-id="48732-115">Affected APIs</span></span>|<ul><li><xref:System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[],System.Int32)?displayProperty=nameWithType></li><li><xref:System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[],System.Int32,System.Threading.CancellationToken)?displayProperty=nameWithType></li><li><xref:System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[],System.TimeSpan)?displayProperty=nameWithType></li></ul>|

